lib/Authentication/AuthTokens/ShibAuthToken.php

```diff
@@ -96,73 +97,67 @@ private function getAttributesInitToken(){
-                 $HTML .= "<div style='text-align: center;'>";
-                 $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                 $HTML .= "</div>";
-                 echo ($HTML);
-                 die();
-            }
-            $this->principal = $_SERVER['voPersonID'];
-            $this->userDetails = array('AuthenticationRealm' => array('EGI Proxy IdP'));
-            return;
-            */
-
-            if( empty($_SERVER['voPersonID'])){// || empty($_SERVER['displayName']) ){
-                die('Did not recieve required attributes from the EGI Proxy Identity Provider to complete authentication, please contact gocdb-admins');
-            }
-            if(empty($_SERVER['entitlement'])){
-                //die('Did not recieve the required entitlement attribute from the EGI Proxy IdP, please contact gocdb-admins');
-                $HTML = '<ul><li>Login requires a GOCDB entitlement value <a href="https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb" target="_blank">https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb</a></li><li>Please, logout or restart your browser and attempt to login again using an identity provider that provides a GOCDB entitlement</li></ul>';
-                $HTML .= "<div style='text-align: center;'>";
-                $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                $HTML .= "</div>";
-                echo ($HTML);
-                die();
-            }
 
-            $entitlementValuesArray = explode(';', $_SERVER['entitlement']);
-            if( !in_array('urn:mace:egi.eu:res:gocdb#aai.egi.eu', $entitlementValuesArray) ){
-                 $HTML = '<ul><li>Login requires a GOCDB entitlement <a href="https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb" target="_blank">https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb</a></li><li>Please, logout or restart your browser and attempt to login again using an identity provider that provides a GOCDB entitlement</li></ul>';
-                 $HTML .= "<div style='text-align: center;'>";
-                 $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                 $HTML .= "</div>";
-                 echo ($HTML);
-                 die();
+        $configService = \Factory::getConfigService();
+        $identityProviders = $configService->getIdentityProvidersInfo();
+
+        foreach ($identityProviders as $provider) {
+            if ($provider['idp'] === $idp) {
+                $name = $provider['name'];
+                $helpUrl = $provider['help_url'];
+
+                if (empty($_SERVER['voPersonID'])) {
+                    die(
+                        "Did not receive required attributes from the "
+                        . "IDP $name to complete authentication. "
```

Comment:
I think this will read better

```suggestion
                        . "$name to complete authentication. "
```

Reply:
Done. 01f3cd379b62c4b3119a25d09f7cef0d852b9904